### PR TITLE
Modify for parallel back ground jobs

### PR DESCRIPTION
**Pull request by kamo-naoyuki**
This is a bash technique: `wait` command always returns 0 without arguments, and if pid is given, then it can return the return code of the background process. 

```
set -e 

fail_command &
wait  # Always 0, so cannot stop here

fail_command &
wait $! # returns non-zero, so this code will exit here.
```

I modified an4/run.sh only for example.

### CODE REVIEW

File: egs/an4/asr1/run.sh [hunk 1]
```diff
@@ -245,15 +246,15 @@ if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
...
-        wait
+            --ctc-weight ${ctc_weight}
 
         score_sclite.sh ${expdir}/${decode_dir} ${dict}
 
     ) &
+    pids+=($!)
     done
-    wait
+    for pid in "${pids[@]}"; do wait ${pid} || { echo "Some jobs failed"; exit 1; }; done
```
Comment (sw005320) [inline egs/an4/asr1/run.sh]: How about `echo "Some background jobs are failed";`
Also, do we need `|| { echo "Some jobs failed"; exit 1; }`?

**Comment by kamo-naoyuki** [comment]
```bash
set -e
pids=()
{ sleep 1 ; echo Job1 ; false; } &
pids+=($!)
{ sleep 3 ; echo Job2 ; false; } &
pids+=($!)
{ sleep 5 ; echo Job3 ; false; } &
pids+=($!)
for i in "${pids[@]}"; do wait ${i}; done
```
I noticed a problem. This program exits at the first waiting, but the other background jobs are still alive, so they are detached from the parent process.

**Comment by kamo-naoyuki** [comment]
This code may be the best practice.

FYI, this one is rejected.

- `jobs -p` instead of storing pids
```bash
<some-command-quickly-finish> &
for pid in $(jobs -p); do wait ${pid}; done  # "jobs -p" possibly returns nothing
```

**Comment by sw005320** [comment]
Thanks! This is one of the issues that I wanted to fix.

**Pull request merged**